ports/esp32/machine_uart.c

```diff
@@ -493,8 +500,27 @@ static void mp_machine_uart_deinit(machine_uart_obj_t *self) {
         vTaskDelete(self->uart_event_task);
         self->uart_event_task = NULL;
     }
-    check_esp_err(uart_driver_delete(self->uart_num));
-    self->uart_queue = NULL;
+    if (self->rxidle_timer != NULL) {
+        machine_timer_disable(self->rxidle_timer);
+    }
```

Comment:
Actually, since this function is also called inside `mp_machine_uart_init_helper`, I'm not sure whether that still works correctly. Let me check.

Reply:
Thanks @DvdGiessen for thinking this through. Maybe we don't need to disable the timer here at all:

* In the case of soft reset, we could deinit timer before uart - so the timer interrupt is already gone.
* In the case of manual deinit then we disable the IRQ, so at most one spurious timer event occurs (IIUC)

Reply:
> In the case of manual deinit then we disable the IRQ, so at most one spurious timer event occurs (IIUC)

Yes, although that event will also propagate (it will call the IRQ handler set by the user), which may be a bit surprising if you deinited the UART but didn't explicitly set the IRQ handler to None first. (Presumably a handler will then try to read data from the UART, raising an OSError.)

Reply:
@projectgus does this point still need addressing?

Reply:
Thanks @DvdGiessen for the clear explanation, you're totally right:

> Otherwise it'd be more consistent to also explicitly remove/disable the IRQ

This was a detail that I'd missed: calling deinit() then init() explicitly will also keep any previously configured irq triggers. Not sure if that's ideal, but makes sense to keep it if it's what we have now.

I've gone with your recommended approach and put a comment there to summarise this reasoning for future reference. Do you mind checking if the comment is accurate, please?

Reply:
Comment seems fine! 👍🏻